sonic_platform_base/sensor_fs.py

```diff
@@ -0,0 +1,159 @@
+"""
+    sensor_fs.py
+
+    File system based sensors implementation. 
+"""
+
+from sonic_platform_base.sensor_base import SensorBase
+from sonic_platform_base.sensor_base import VoltageSensorBase
+from sonic_platform_base.sensor_base import CurrentSensorBase
+import logging
+
+
+class SensorFs(SensorBase):
+    """Implementation of file system based sensor class"""
+
+    __defaults = {
+        'name': '',
+        'sensor': '',
+        'high_thresholds': ['N/A', 'N/A', 'N/A'],
+        'low_thresholds': ['N/A', 'N/A', 'N/A'],
+        'position': -1,
+    }
+
+    def __init__(self, sensor_type='sensor', **kw):
+
+        super(SensorFs, self).__init__()
+
+        if 'sensor' not in kw:
+            raise Exception('Failed to initialize sensor')
```

Comment:
I suppose some daemon will call this class, will the daemon catch this Exception? What is the expected behavior for the daemon when hits this exception?

Reply:
Daemon is Sensormond. It will continue and not process this sensor. Look at the voltage sensor factory function below which is instantiating the sensors.